### ISSUE TITLE
[Quality] Standardize Parameter Naming Across API

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ProjectOdyssey#2643]
## Problem

Inconsistent parameter naming makes API harder to learn and use.

## Common Inconsistencies

**Dimension parameter:**
```mojo
fn sum(tensor: ExTensor, axis: Int)      # Uses "axis"
fn mean(tensor: ExTensor, dim: Int)      # Uses "dim"
fn max(tensor: ExTensor, dimension: Int) # Uses "dimension"
```

**Input parameter:**
```mojo
fn relu(input: ExTensor)   # Uses "input"
fn tanh(tensor: ExTensor)  # Uses "tensor"
fn sigmoid(x: ExTensor)    # Uses "x"
```

**Output parameter:**
```mojo
fn forward(input: ExTensor) -> ExTensor  # Returns unnamed
fn backward(...) -> grad: ExTensor       # Returns named
```

## Proposed Standards

**For reduction operations:**
- Use `axis: Int` (consistent with NumPy/PyTorch)

**For unary operations:**
- Use `tensor: ExTensor` (clear and descriptive)

**For binary operations:**
- Use `a: ExTensor, b: ExTensor` or `left: ExTensor, right: ExTensor`

**For neural network layers:**
- Use `input: ExTensor` for forward pass
- Use `grad_output: ExTensor` for backward pass

**For optional parameters:**
- Use `keepdim: Bool = False` (not `keep_dim` or `keep_dims`)
- Use `inplace: Bool = False` (not `in_place`)

## Audit Needed

1. List all public API functions
2. Document current parameter names
3. Identify inconsistencies
4. Propose standard naming
5. Create migration plan

## Impact

- **Severity**: LOW
- **API consistency**: Easier to learn
- **User experience**: Predictable parameter names

**Comment by mvillmow** [comment]
## Detailed Implementation Plan

Limit reached · resets 2pm (America/Los_Angeles)

**Comment by mvillmow** [comment]
## Dependencies (AI-Analyzed)

**Priority**: P1

### Depends On
- #2642

### Blocks
- #2645
- #2654

### Analysis
Parameter naming standardization requires API to be validated (#2642). Blocks API reference documentation (#2645) and contributing guidelines (#2654).

---
*Analyzed by Claude Code dependency analyzer*
